crates/containerd-shim-wasm/src/sandbox/shim.rs

```diff
@@ -1,1274 +0,0 @@
-        debug!("call prehook before the start");
-        oci::setup_prestart_hooks(spec.hooks())?;
-
-        Ok(api::CreateTaskResponse {
-            pid: std::process::id(),
-            ..Default::default()
-        })
-    }
-
-    fn task_start(&self, req: api::StartRequest) -> Result<api::StartResponse> {
-        if req.exec_id().is_empty().not() {
-            return Err(ShimError::Unimplemented("exec is not supported".to_string()).into());
-        }
-
-        let i = self.get_instance(req.id())?;
-        let pid = i.start()?;
-
-        let mut pid_w = i.pid.write().unwrap();
-        *pid_w = Some(pid);
-        drop(pid_w);
-
-        self.send_event(TaskStart {
-            container_id: req.id().into(),
-            pid,
-            ..Default::default()
-        });
-
-        let sender = self.events.clone();
-        let id = req.id().to_string();
-
-        thread::Builder::new()
-            .name(format!("{}-wait", req.id()))
-            .spawn(move || {
-                let exit_code = i.wait();
-
-                let timestamp = new_timestamp().unwrap();
-                let event = TaskExit {
-                    container_id: id.clone(),
-                    exit_status: exit_code.0,
-                    exited_at: MessageField::some(timestamp),
-                    pid,
-                    id,
-                    ..Default::default()
-                };
-
-                let topic = event.topic();
-                sender
-                    .lock()
-                    .unwrap()
-                    .send((topic.clone(), Box::new(event)))
-                    .unwrap_or_else(|err| {
-                        error!("failed to send event for topic {}: {}", topic, err)
-                    });
-            })
-            .context("could not spawn thread to wait exit")
-            .map_err(Error::from)?;
-
-        debug!("started: {:?}", req);
-
-        Ok(api::StartResponse {
-            pid,
-            ..Default::default()
-        })
-    }
-
-    fn task_kill(&self, req: api::KillRequest) -> Result<()> {
-        if req.exec_id().is_empty().not() {
-            return Err(Error::InvalidArgument("exec is not supported".to_string()));
-        }
-        self.get_instance(req.id())?.kill(req.signal())?;
-        Ok(())
-    }
-
-    fn task_delete(&self, req: api::DeleteRequest) -> Result<api::DeleteResponse> {
-        if req.exec_id().is_empty().not() {
-            return Err(Error::InvalidArgument("exec is not supported".to_string()));
-        }
-
-        let i = self.get_instance(req.id())?;
-
-        i.delete()?;
-
-        let pid = i.pid.read().unwrap().unwrap_or_default();
-
-        let mut event = TaskDelete {
-            container_id: req.id().into(),
-            pid,
-            ..Default::default()
-        };
-
-        let mut resp = api::DeleteResponse {
-            pid,
-            ..Default::default()
-        };
-
-        if let Some(ec) = i.wait_timeout(Duration::ZERO) {
```

Comment:
I don't see this block of logic in the new code. Why is it removed?

Reply:
It's not removed. This logic is captured by:
* converting the timestamps is [here](https://github.com/containerd/runwasi/pull/345/files#diff-9aeebb54e7b1d3667456ff20b60bd3416f3e44f11866f6f2821282f626177606R251).
* setting the event's s`exit_status` and `exited_at` is [here](https://github.com/containerd/runwasi/pull/345/files#diff-9aeebb54e7b1d3667456ff20b60bd3416f3e44f11866f6f2821282f626177606R258-R259)
* setting the response `exit_status` and `exited_at` is [here](https://github.com/containerd/runwasi/pull/345/files#diff-9aeebb54e7b1d3667456ff20b60bd3416f3e44f11866f6f2821282f626177606R265-R266)